fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1192,32 +1194,56 @@ public RecordCursor<FDBStoredRecord<Message>> scanRecords(@Nullable final Tuple
     }
 
     @Nonnull
-    @SuppressWarnings("PMD.CloseResource")
     public <M extends Message> RecordCursor<FDBStoredRecord<M>> scanTypedRecords(@Nonnull RecordSerializer<M> typedSerializer,
                                                                                  @Nullable final Tuple low, @Nullable final Tuple high,
                                                                                  @Nonnull final EndpointType lowEndpoint, @Nonnull final EndpointType highEndpoint,
                                                                                  @Nullable byte[] continuation,
                                                                                  @Nonnull ScanProperties scanProperties) {
+        return scanTypedRecordsInternal(typedSerializer,
+                                        builder -> builder.setLow(low, lowEndpoint).setHigh(high, highEndpoint),
+                                        continuation, scanProperties);
+    }
+
+    @Nonnull
+    @Override
+    public RecordCursor<FDBStoredRecord<Message>> scanRecordsKeyRange(@Nonnull final KeyRange range, @Nullable final byte[] continuation, @Nonnull final ScanProperties scanProperties) {
+        return scanTypedRecordsKeyRange(serializer, range, continuation, scanProperties);
+    }
+
+    @Nonnull
+    public <M extends Message> RecordCursor<FDBStoredRecord<M>> scanTypedRecordsKeyRange(@Nonnull RecordSerializer<M> typedSerializer,
+                                                                                         @Nonnull final KeyRange range,
+                                                                                         @Nullable byte[] continuation,
+                                                                                         @Nonnull ScanProperties scanProperties) {
+        return scanTypedRecordsInternal(typedSerializer,
+                                        builder -> builder.setRange(range),
+                                        continuation, scanProperties);
+    }
+
+    @Nonnull
+    @SuppressWarnings("PMD.CloseResource")
+    private <M extends Message> RecordCursor<FDBStoredRecord<M>> scanTypedRecordsInternal(@Nonnull RecordSerializer<M> typedSerializer,
+                                                                                          @Nonnull Consumer<KeyValueCursor.Builder> setRange,
```

Comment:
minor: I was trying to think if there was a better way to structure this than a `Consumer`. The answer may be "no". The best alternatives I could think of would be to either have this take a `KeyValueCursor.Builder` than callers would pre-fill with the range already set, or something more exotic like adding a `.setRange(KeyValueCursor.Builder)` to the `TupleRange` and `KeyValueRange` abstractions, or making `TupleRange` and `KeyValueRange` implement some common interface that could set the low and high endpoints on a `KeyValueCursor.Builder`.